orchestrator/db/filters/process.py

```diff
@@ -11,152 +11,97 @@
-    BASE_CAMEL
-    | BASE_SNAKE
-    | {
-        "isTask": generic_bool_filter(ProcessTable.is_task),
-        "is_task": generic_bool_filter(ProcessTable.is_task),
-        "assignee": generic_values_in_column_filter(ProcessTable.assignee),
-        "lastStatus": generic_values_in_column_filter(ProcessTable.last_status),
-        "product": product_filter,
-        "productTag": tag_filter,
-        "tag": tag_filter,
-        "subscriptions": subscriptions_filter,
-        "subscriptionId": subscription_id_filter,
-        "subscription_id": subscription_id_filter,
-        "target": target_filter,
-        "workflow_name": workflow_name_filter,
-        "workflowName": workflow_name_filter,
-        "customer": customer_filter,
-        "istask": generic_bool_filter(ProcessTable.is_task),  # TODO: will be removed in 1.4
-        "status": generic_values_in_column_filter(ProcessTable.last_status),  # TODO: will be removed in 1.4
-        "last_status": generic_values_in_column_filter(ProcessTable.last_status),  # TODO: will be removed in 1.4
-        "creator": generic_is_like_filter(ProcessTable.created_by),  # TODO: will be removed in 1.4
-    }
-)
 
+    return ProcessTable.process_id.in_(process_workflow)
 
-def make_product_clause(filter_generator: WhereCondGenerator) -> WhereCondGenerator:
-    """The passed filter_generator takes a Node and returns a where clause acting on a ProductTable column."""
 
-    def product_clause(node: Node) -> BinaryExpression:
-        process_subscriptions = (
-            select(ProcessSubscriptionTable.process_id)
-            .join(SubscriptionTable)
-            .join(ProductTable)
-            .where(filter_generator(node))
-            .subquery()
-        )
-        return ProcessTable.process_id.in_(process_subscriptions)
-
-    return product_clause
+def workflow_targets_clause(node: Node) -> BinaryExpression:
+    process_workflows = (
+        select(ProcessTable.process_id)
+        .join(WorkflowTable)
+        .where(WorkflowTable.deleted_at.is_(None), WorkflowTable.target == node_to_str_val(node))
+        .subquery()
+    )
+    return ProcessTable.process_id.in_(process_workflows)
 
 
 PROCESS_TABLE_COLUMN_CLAUSES = default_inferred_column_clauses(ProcessTable) | {
+    "customer": customer_clause,
     "product": make_product_clause(inferred_filter(ProductTable.name)),
     "product_description": make_product_clause(inferred_filter(ProductTable.description)),
+    "subscription_id": make_subscription_id_clause(inferred_filter(SubscriptionTable.subscription_id)),
     "tag": make_product_clause(filter_exact(ProductTable.tag)),
+    "target": workflow_targets_clause,
+    "workflow_name": workflow_name_clause,
 }
 
 
-process_filter_fields = list(PROCESS_FILTER_FUNCTIONS_BY_COLUMN.keys())
-filter_processes = generic_filter(PROCESS_FILTER_FUNCTIONS_BY_COLUMN)
+process_filter_fields = list(PROCESS_TABLE_COLUMN_CLAUSES.keys())
```

Comment:
This should be a function to get filter fields, otherwise these keys will never from the initial `PROCESS_TABLE_COLUMN_CLAUSES ` keys, extra filters from projects that use this package will never be added to the list.

```suggestion
def get_process_filter_fields() -> list[str]:
    return list(PROCESS_TABLE_COLUMN_CLAUSES.keys())
```

should be changed everywhere

Reply:
Shall we change it for all filter and sort lists for all resources?

Reply:
Done for all filter/sorting field lists

Reply:
nice!